src/coprocessor/dag/aggr_fn/impl_count.rs

```diff
@@ -119,26 +119,14 @@ impl<T: Evaluable> super::AggrFunctionStateUpdatePartial<T> for AggrFnStateCount
     }
 }
 
-impl<T> super::AggrFunctionStateResultPartial<T> for AggrFnStateCount
-where
-    T: super::AggrResultAppendable + ?Sized,
-{
+impl super::AggrFunctionState for AggrFnStateCount {
     #[inline]
-    default fn push_result(&self, _ctx: &mut EvalContext, _target: &mut T) -> Result<()> {
-        panic!("Unmatched result append target type")
-    }
-}
-
-impl super::AggrFunctionStateResultPartial<Vec<Option<Int>>> for AggrFnStateCount {
-    #[inline]
-    fn push_result(&self, _ctx: &mut EvalContext, target: &mut Vec<Option<Int>>) -> Result<()> {
-        target.push(Some(self.count as Int));
+    fn push_result(&self, _ctx: &mut EvalContext, target: &mut [VectorValue]) -> Result<()> {
+        target[0].push(Some(self.count as Int));
```

Comment:
Is necessary to add an assert to here, like `impl_avg:L159`?

Reply:
Maybe. I can add it in future.